tests/unit/processing/test_dataset_metadata_update.py

```diff
@@ -0,0 +1,526 @@
+import json
+import os
+import tempfile
+from shutil import copy2
+from unittest.mock import Mock, patch
+
+import pytest
+import scanpy
+import tiledb
+from parameterized import parameterized
+from rpy2.robjects.packages import importr
+
+from backend.common.utils.corpora_constants import CorporaConstants
+from backend.common.utils.cxg_generation_utils import convert_dictionary_to_cxg_group
+from backend.layers.common.entities import (
+    CollectionVersionId,
+    DatasetArtifactMetadataUpdate,
+    DatasetArtifactType,
+    DatasetConversionStatus,
+    DatasetProcessingStatus,
+    DatasetStatusKey,
+    DatasetUploadStatus,
+    DatasetValidationStatus,
+    DatasetVersionId,
+)
+from backend.layers.processing.dataset_metadata_update import DatasetMetadataUpdater
+from backend.layers.thirdparty.s3_provider_mock import MockS3Provider
+from tests.unit.backend.fixtures.environment_setup import fixture_file_path
+from tests.unit.backend.layers.common.base_test import DatasetArtifactUpdate, DatasetStatusUpdate
+from tests.unit.processing.base_processing_test import BaseProcessingTest
+
+base = importr("base")
+seurat = importr("SeuratObject")
+
+
+def mock_apply_async(func, args=()):
+    return Mock(return_value=func(*args))
+
+
+mock_enter = Mock(apply_async=Mock(side_effect=mock_apply_async))
+mock_pool = Mock(__enter__=Mock(return_value=mock_enter), __exit__=Mock())
+mock_pool_constructor = Mock(return_value=mock_pool)
+
+
+@patch("backend.layers.processing.dataset_metadata_update.multiprocessing.Pool", mock_pool_constructor)
```

Comment:
If only some of the tests needs this mock, pull those tests out into their own test class. I don't think update_h5ad, update_cxg, or update_seurat need it.

Reply:
I am tempted to keep it given that a majority of the tests need it, and it doesn't negatively impact those that don't. But I can be swayed if you think there is a downside

Reply:
Oh nvm I misread your comment--yeah I can put those in a separate test class